ftml/src/preproc/typography.rs

```diff
@@ -107,13 +108,19 @@ impl Replacer {
                     replacement,
                 );
 
-                while let Some(capture) = regex.captures(text) {
+                let mut offset = 0;
+                let repl_len = replacement.len();
+
+                while let Some(capture) = regex.captures_at(text, offset) {
                     let range = {
-                        let mtch = capture
+                        let full_match = capture
                             .get(0)
                             .expect("Regular expression lacks a full match");
+                        let mtch = capture.name("repl").or(Some(full_match)).unwrap();
 
-                        mtch.start()..mtch.end()
+                        offset = mtch.start() + repl_len;
```

Comment:
Since `repl_len` is only used once, we can get rid of the separate variable and just use:
```suggestion
                        offset = mtch.start() + replacement.len();
```